mycli/main.py

```diff
@@ -166,11 +170,16 @@ def read_my_cnf_files(self, files, keys):
                 cnf.merge(e.config)
                 pass
 
+        sections = ['client']
+        if self.defaults_suffix:
+            sections.append('client{0}'.format(self.defaults_suffix))
+
         def get(key):
-            try:
-                return cnf['client'][key]
-            except KeyError:
-                return None
+            result = None
+            for sect in sections:
+                if sect in cnf and key in cnf[sect]:
+                    result = cnf[sect][key]
```

Comment:
Only just saw the email. No, the values from files lower on the list should overrides previous values.